cql3/statements/select_statement.cc

```diff
@@ -404,32 +405,58 @@ select_statement::do_execute(query_processor& qp,
 
     auto key_ranges = _restrictions->get_partition_key_ranges(options);
 
+    bool send_tablet = false;
+
+    auto token = dht::token();
+    unsigned shard;
+
+    if (key_ranges.size() > 0 && query::is_single_partition(key_ranges.front())) {
+        token = key_ranges[0].start()->value().as_decorated_key().token();
+        shard = _schema->table().shard_of(token);
+    }
+
+    auto&& table = qp.proxy().local_db().find_column_family(_schema);
+    auto erm = table.get_effective_replication_map();
+    auto tablet_replicas = erm->get_replicas_for_reading(token);
+    auto token_range = erm->get_token_range(token);
+
+    if ((!tablet_replicas.empty() && erm->get_token_metadata_ptr()->get_endpoint_for_host_id(tablet_replicas.front().host) != utils::fb_utilities::get_broadcast_address())
```

Comment:
Why do you expect local host to be in the front of the replica list? I think it doesn't have to. The driver should alternate between hosts.

Reply:
I followed this: https://github.com/scylladb/scylladb/blob/master/service/storage_proxy.cc#L5300-L5304, but maybe I understand something wrong

Reply:
That piece relies on storage_proxy::get_endpoints_for_reading() calling sort_endpoints_by_proximity(). But here we can't, because we didn't sort the endpoints yet.

Reply:
Then, how can I check is the read is local?

Reply:
The read is local if local host is any of tablet_replicas. We can add a helper for that:

```
bool contains(const tablet_replicas&, host_id);
```

Reply:
Changed